src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -0,0 +1,2164 @@
+                            {
+                                WinHttpException.ThrowExceptionUsingLastError();
+                            }
+                        }
+                    }
+
+                    state.RequestMessage.RequestUri = redirectUri;
+
+                    // For security reasons, we drop the server credential if it is a 
+                    // NetworkCredential.  But we allow credentials in a CredentialCache
+                    // since they are specifically tied to URI's.
+                    if (!(state.ServerCredentials is CredentialCache))
+                    {
+                        state.ServerCredentials = null;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Interop.WinHttp.WinHttpCloseHandle(handle);
+                    state.SavedException = ex;
+                }
+            }
+            else if (internetStatus == Interop.WinHttp.WINHTTP_CALLBACK_STATUS_SENDING_REQUEST)
+            {
+                if (state.RequestMessage.RequestUri.Scheme != UriSchemeHttps)
+                {
+                    // Not SSL/TLS.
+                    return;
+                }
+
+                // Grab the channel binding token (CBT) information from the request handle and put it into
+                // the TransportContext object.
+                try
+                {
+                    state.TransportContext.SetChannelBinding(state.RequestHandle);
+                }
+                catch (Exception ex)
+                {
+                    Interop.WinHttp.WinHttpCloseHandle(handle);
+                    state.SavedException = ex;
+
+                    return;
+                }
+
+                if (state.ServerCertificateValidationCallback != null)
+                {
+                    IntPtr data = IntPtr.Zero;
+                    uint dataSize = (uint)IntPtr.Size;
+
+                    if (Interop.WinHttp.WinHttpQueryOption(
+                        state.RequestHandle,
+                        Interop.WinHttp.WINHTTP_OPTION_SERVER_CERT_CONTEXT,
+                        ref data,
+                        ref dataSize))
+                    {
+                        var serverCertificate = new X509Certificate2(data);
+                        Interop.Crypt32.CertFreeCertificateContext(data);
+
+                        X509Chain chain = null;
+                        SslPolicyErrors sslPolicyErrors;
+
+                        try
+                        {
+                            CertificateHelper.BuildChain(
+                                serverCertificate,
+                                state.RequestMessage.RequestUri.Host,
+                                state.CheckCertificateRevocationList,
+                                out chain,
+                                out sslPolicyErrors);
+
+                            var result = state.ServerCertificateValidationCallback(
```

Comment:
`var result` here did set off my `var` alarm.  I expected a rich object, and it either *is* a rich object with an `implicit operator true`, or it's just a `bool`.  I'm greatly leaning toward `bool`... but it made me double-take.